File: backend/dotnet/Models/Team.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace dotnet.Models
{
    [Table("teams")]
    public class Team
    {
        [Key]
        [Column("team_id")]
        public String Id { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; } = string.Empty;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [JsonIgnore]
        public ICollection<TeamMember> TeamMembers { get; set; } = new List<TeamMember>();
        [JsonIgnore]
        public ICollection<Project> Projects { get; set; } = new List<Project>();
    }
}

File: backend/golang/internal/model/team.go
package model

import "time"

type Team struct {
	ID        uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	Name      string    `json:"name" gorm:"not null"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`

	Members  []TeamMember `gorm:"foreignKey:TeamID" json:"teamMembers"`
	Projects []Project    `gorm:"foreignKey:TeamID" json:"projects"`
}
